README.md

```diff
@@ -18,48 +18,123 @@ Answer the following data queries. Keep track of the SQL you write by pasting it
+```
+
 ### add a customer record for _"The Shire"_, the contact name is _"Bilbo Baggins"_ the address is _"1 Hobbit-Hole"_ in _"Bag End"_, postal code _"111"_ and the country is _"Middle Earth"_.
 > This can be done with the INSERT INTO clause
 
+```
+insert
+into Customers(CustomerName, ContactName, Address, City, PostalCode, Country)
+values('the shire', 'bilbo baggins', '1-hobbit hole', 'bag end', '111', 'middle earth')
+```
+
+
 ### update _Bilbo Baggins_ record so that the postal code changes to _"11122"_.
 > This can be done with UPDATE and WHERE clauses
 
+```
+update Customers
+set PostalCode = '11122'
+where CustomerName = 'the shire'
+```
+
+
 ### list orders grouped by customer showing the number of orders per customer. _Rattlesnake Canyon Grocery_ should have 7 orders.
 > This can be done with SELECT, COUNT, JOIN and GROUP BY clauses. Your count should focus on a field in the Orders table, not the Customer table
-
 > There is more information about the COUNT clause on [W3 Schools](https://www.w3schools.com/sql/sql_count_avg_sum.asp)
 
+```
+select c.customername as CustomerName, count(o.customerid) as TotalOrders
+from customers as c
+INNER join orders as o
+on c.customerid = o.customerid
+group by c.customername
+```
+
 ### list customers names and the number of orders per customer. Sort the list by number of orders in descending order. _Ernst Handel_ should be at the top with 10 orders followed by _QUICK-Stop_, _Rattlesnake Canyon Grocery_ and _Wartian Herkku_ with 7 orders each.
 > This can be done by adding an ORDER BY clause to the previous answer
 
+```
+select c.customername as CustomerName, count(o.customerid) as TotalOrders
+from customers as c
+INNER join orders as o
+on c.customerid = o.customerid
+group by c.customername
+order by count(o.customerid) desc
+```
+
 ### list orders grouped by customer's city showing number of orders per city. Returns 58 Records with _Aachen_ showing 2 orders and _Albuquerque_ showing 7 orders.
 > This is very similar to the previous two queries, however, it focuses on the City rather than the CustomerName
 
+```
+select c.city as CityName, count(o.customerid) as TotalOrders
+from customers as c
+INNER join orders as o
+on c.customerid = o.customerid
+group by city
+order by count(o.customerid) asc /* remove this line for a normal return */ 
+```
+
+
 ### delete all customers that have no orders. Should delete 17 (or 18 if you haven't deleted the record added) records.
 > This is done with a DELETE query
-
 > In the WHERE clause, you can provide another list with an IN keyword this list can be the result of another SELECT query. Write a query to return a list of CustomerIDs that meet the criteria above. Pass that to the IN keyword of the WHERE clause as the list of IDs to be deleted
- 
 > Use a LEFT JOIN to join the Orders table onto the Customers table and check for a NULL value in the OrderID column
 
+```
+delete from customers
+where customerid not in (select customerid from orders)
+```
```

Comment:
Just as an extra tidbit, there are other ways to perform this operation - here are a couple of them:

```
DELETE
FROM customers
WHERE customerid IN (
    SELECT c.customerid
    FROM customers c LEFT JOIN orders o
    ON o.CustomerID = c.CustomerID  
    WHERE o.orderid IS NULL
)
```
Following the *exact* directions given for this particular request, this is query that you would end up with.

```
DELETE FROM Customers c
WHERE NOT EXISTS (
    SELECT *
    FROM Orders o
    WHERE c.CustomerID=o.CustomerID
)
```
For each customer, this query checks whether the customer's ID matches a `CustomerID` entry in the `Orders` table, and if it doesn't find a match (hence the [`EXISTS`](https://www.w3schools.com/sql/sql_exists.asp) keyword used with the `NOT` operator), it will delete the customer entry from the `Customers` table.